pkg/csi/cinder/nodeserver.go

```diff
@@ -526,15 +526,20 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
 	}
 	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume path not provided")
+	}
 
-	args := []string{"-o", "source", "--first-only", "--noheadings", "--target", volumePath}
-	output, err := ns.Mount.Mounter().Exec.Command("findmnt", args...).CombinedOutput()
+	_, err := ns.Cloud.GetVolume(volumeID)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not determine device path: %v", err)
-
+		if cpoerrors.IsNotFound(err) {
```

Comment:
@jichenjc Looks to me this code shouldn't be required. From the issue, I see that the error should come from  https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L716 here, which is nodeGetVolumeStats . So code should work irrespective of this.

Reply:
@ramineni you should check 4.1.0 code , not master branch...
I can't point you the 4.1.0 code but if you checkout 4.1.0 

you will see line 716 is following
```
It("should fail when volume is not found", func() {
                        _, err := r.NodeExpandVolume(
                                context.Background(),
                                &csi.NodeExpandVolumeRequest{
                                        VolumeId:   sc.Config.IDGen.GenerateUniqueValidVolumeID(),
                                        VolumePath: "some/path",
                                },
                        )
                        Expect(err).To(HaveOccurred())

                        serverError, ok := status.FromError(err)
                        Expect(ok).To(BeTrue())
                        Expect(serverError.Code()).To(Equal(codes.NotFound))
                })
```

Reply:
@jichenjc ok got it.